graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -341,7 +347,22 @@ abstract class BaseDataTypeGenerator(
             constructorBuilder
                 .addParameter(parameterBuilder.build())
                 .addModifiers(Modifier.PUBLIC)
-                .addStatement("this.\$N = \$N", ReservedKeywordSanitizer.sanitize(it.name), ReservedKeywordSanitizer.sanitize(it.name))
+        }
+
+        fieldDefinitions.forEach {
+            val constructor = constructorBuilder
+                .addStatement(
+                    "this.\$N = \$N",
+                    ReservedKeywordSanitizer.sanitize(it.name),
+                    ReservedKeywordSanitizer.sanitize(it.name)
+                )
+            if (!(!it.isNullable && it.initialValue == null)) {
```

Comment:
It's probably cleaner to write this as:

```kotlin
if (it.isNullable && it.initialValue == null) {
   // ...
}
```

Reply:
@kilink Fields are generated for all these cases: 
1. Nullable | No default values --> Generate fields
2. Nullable | With default values --> Generate fields
3. Non-Nullable | No default values --> Generate fields
4. Non-Nullable | With default values --> No need for boolean fields

Fields are generated for both nullable field with default value as well as non-nullable with default value might get assigned default values by initializer without direct request from the client. This is our understanding from existing graphql services leveraging sparse update solution. Having Boolean field for other cases will help identify values set by the client. 

With this, the I can update the double negative to simplify the logic: 

`if (it.isNullable && it.initialValue != null)) `

Is this acceptable?

Reply:
There's no reason to generate fields for any of the cases except the first one: a nullable field with no defaults. For any case with a default value, it's impossible for DGS to tell if the value was explicitly supplied or not, graphql-java will hand us the variables with the default value already there. So the flag here would always be "true". The flag is pretty useless in that case and misleading.

Reply:
Ack, updated to generate fields for nullable fields with no default values.